test/test.ts

```diff
@@ -1,23 +1,74 @@
-import axios from "axios";
-import server from "../src/index"; // Import the server setup
 import { describe, before, it, after } from "mocha"; // Mocha imports
-import assert from "assert";
+import { expect, assert } from "chai";
+import { setup } from "../src/setup";
+import { prisma } from "../src/setup-db";
+import { server, url } from "../src/setup-server";
+import axios from "axios";
 
 describe("GraphQL Server Test", function () {
-  let serverUrl: string;
-
   before(async function () {
-    serverUrl = "http://localhost:4000";
+    console.log("Starting setup");
+    await setup();
+    console.log("Setup complete");
   });
 
-  it("should return 'Hello world!' from the hello query", async function () {
-    const response = await axios.post(serverUrl, {
+  it("Returned 'Hello world!' from the hello query", async function () {
+    const response = await axios.post(url, {
       query: "{ hello }"
     });
 
     assert.equal(response.data.data.hello, "Hello world!");
   });
 
+  it("Created a new user with a mutation", async function () {
+    const response = await axios.post(url, {
+      query: `
+          mutation {
+            createUser(data: {
+              name: "User Test",
+              email: "test@example.com",
+              password: "123abc",
+              birthDate: "01-01-2000"
+            }) {
+              id
+              name
+              email
+              birthDate
+            }
+          }
+        `
+    });
+
+    expect(response.data).to.have.property("data");
+    expect(response.data.data).to.have.property("createUser");
+    expect(response.data.data.createUser).to.not.be.null;
+    expect(response.data.data.createUser.id).to.not.be.null;
+    assert.equal(response.data.data.createUser.name, "User Test");
+    assert.equal(response.data.data.createUser.email, "test@example.com");
+    assert.equal(response.data.data.createUser.birthDate, "01-01-2000");
+  });
+
+  it("Found the newly created user", async function () {
+    const createdUser = await prisma.user.findUnique({
+      where: {
+        email: "test@example.com"
+      }
+    });
+    expect(createdUser).to.not.be.null;
+    if (createdUser) {
+      expect(createdUser.email).to.be.equal("test@example.com");
+    }
+  });
+
+  it("Deleted the newly created user", async function () {
+    const deletedUser = await prisma.user.delete({
+      where: {
+        email: "test@example.com"
+      }
+    });
+    assert.equal(deletedUser.email, "test@example.com");
+  });
```

Comment:
Aqui acho que vale citar que nós organizamos os testes de uma forma em que cada caso de teste nosso verifique como se comporta uma mesma query/mutation em condições diferentes. Só que esses 2 casos de teste aqui não fazem exatamente isso. O primeiro apenas consulta no banco se tem um usuário com aquele e-mail, e o segundo é um "caso de teste" que só deleta o usuário (não está testando algo, portanto).

Resumindo: cada `it` deveria fazer todo o ciclo de:
1. Preparar o banco (se necessário)
2. Fazer a query ou a mutation que estamos testando
3. Checar os resultados (response da query e/ou estado do banco)
4. Limpar o banco

Aqui vc parece ter feito esse processo de forma separada, em its diferentes.